### DISCUSSION TITLE
Terrain modeling and wake induced turbulence

### DISCUSSION THREAD

**Discussion by snakefeifei855** [NatLabRockies/floris#501] [Q&A] [answered]
I noticed that the coordinates of the layout are only x and y, Is Floris the default that all wind farms are flat terrain?? Because if the altitudes (z) of the two turbines are too different, there may be no wake effects, so considering only x and y will not apply.

Also, can Floris calculate added turbulence and wind speed with the wake effect?

**Comment by Bartdoekemeijer** [comment, marked as answer]
Hi @snakefeifei855,

Indeed, FLORIS assumes a flat terrain. You can "fake" a complex terrain by assigning turbines a different hub height. You can read more about this in #224, though be aware the example code therein was written for FLORIS v2.4 and the API has changed since then. Also, yes, FLORIS does calculate the added turbulence and the wind speeds in the wake. You can check out [example 02](https://github.com/NREL/floris/blob/main/examples/02_visualizations.py) on calculating and visualizing the wind speeds in wakes. The added turbulence is only calculated at turbine rotors, and currently cannot yet be calculated at any arbitrary point in the farm. You can access the turbulence intensity values at `fi.floris.flow_field.turbulence_intensity_field`. If you want to calculate the added turbulence intensity at a particular point, you can add a turbine to floris located at that point and get the value from `fi.floris.flow_field.turbulence_intensity_field`, in a similar approach as #418. It is slow, but it's an easy workaround for the time being.

Let me know if you have any other questions.

Bart
